inc/ThirdParty/Plugins/PageBuilder/Elementor.php

```diff
@@ -161,4 +142,31 @@ public function add_fix_animation_script( $html ) {
 
 		return $html;
 	}
+
+	/**
+	 * Excludes JS files from minify/combine JS
+	 *
+	 * @since 3.10.9
+	 *
+	 * @param array $excluded_files Array of excluded patterns.
+	 *
+	 * @return array
+	 */
+	public function exclude_js( $excluded_files ): array {
+		if ( ! $this->options->get( 'minify_concatenate_js', false ) ) {
+			return $excluded_files;
+		}
+
+		if ( ! $this->options->get( 'cache_logged_user', false ) ) {
+			return $excluded_files;
+		}
+
+		if ( ! is_user_logged_in() ) {
+			return $excluded_files;
+		}
+
+		$excluded_files[] = '/wp-includes/js/dist/hooks.min.js';
```

Comment:
Do we need to handle both `min` and source files? as if `SCRIPT_DEBUG` is true, the page will load the source file not the minified one.
Like what we do here: https://github.com/wp-media/wp-rocket/blob/f7667cd723f29484c222d49c274fda6a11abffd5/inc/Engine/Optimization/DeferJS/DeferJS.php#L231

Reply:
Indeed it's better to cover both, code is updated